qiskit/pulse/timeslots.py

```diff
@@ -281,8 +295,9 @@ def _merge_timeslot(self, timeslot: Timeslot) -> int:
             if interval.start >= ch_interval.stop:
                 break
             elif interval.has_overlap(ch_interval):
-                raise PulseError("Timeslot: {0} overlaps with existing"
-                                 "Timeslot: {1}".format(timeslot, ch_timeslot))
+                overlap_start, overlap_end = interval.get_overlap(ch_interval)
```

Comment:
```suggestion
                overlap_start = self.start if self.start > interval.start else interval.start
                overlap_stop = interval.stop if self.stop > interval.stop else self.stop
```
Rather than a new method, let's just keep the code here for now. I do appreciate that a separate func is easier to test, but in this case I prefer adding 1 line over adding 14 lines, and we don't use `get_overlap` anywhere else yet. I think it's possible to grab the error message that is returned in tests -- you can check that the message has the right values.